### PR TITLE
Use more Rust idioms

### PR DESCRIPTION
**Pull request by dancrossnyc**
Use Monad-like operations on Option<T> and Result<T, E> where applicable; use some of the iterator methods for e.g. searching, generally clean up a bit.

### CODE REVIEW

File: src/apcb.rs [hunk 1]
```diff
@@ -293,41 +277,34 @@ impl<'a> ApcbIterMut<'a> {
...
-                    return Err(Error::FileSystem(
-                        FileSystemError::InconsistentHeader,
-                        "GROUP_HEADER::group_size",
-                    ));
-                }
-                self.remaining_used_size -= group_size;
-                Ok(e)
-            }
-            Err(e) => Err(e),
+        assert!(self.remaining_used_size != 0, "Internal error");
```
Comment (daym) [inline src/apcb.rs:297]: Can these asserts be disabled? I would not like to corrupt anything.

File: Cargo.toml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 name = "amd-apcb"
 version = "0.3.0"
 authors = ["Oxide Computer"]
-edition = "2018"
+edition = "2021"
```
Comment (daym) [inline Cargo.toml:5]: This causes an extra warning that TryFrom now is auto-imported.

**Review by daym — CHANGES_REQUESTED** [review]
Maybe also fix the TryFrom import warning. Otherwise looks great!

File: src/apcb.rs (hunk 1 above)
Comment (dancrossnyc) [reply src/apcb.rs:297]: No.  They run in both debug and release modes, unconditionally.

From https://doc.rust-lang.org/std/macro.assert.html:
> Assertions are always checked in both debug and release builds, and cannot be disabled. See [debug_assert!](https://doc.rust-lang.org/std/macro.debug_assert.html) for assertions that are not enabled in release builds by default.

File: Cargo.toml (hunk 2 above)
Comment (dancrossnyc) [reply Cargo.toml:5]: Fixed.

**Review by daym — APPROVED** [review]

**Pull request merged**